attempt to improve documentation

Kavindu-Dodan opened the pull request:
## This PR

Attempts to improve OFO documentation

agardnerIT reviewed (changes requested):
Thanks for these changes. They largely look great! A few comments though:

1. `docs/getting_started.md` and `docs/quick_start.md` should be combined. They serve largely the same purpose and as such, having both is confusing and redundant.
2. At the end of `docs/installation.md` we should link to `docs/getting_started.md` otherwise, after installation, I don't know what else to do.
3. At the end of `docs/quick_start.md` (or wherever that content ends up, we should mention that the OFO uses flagd as a flag provider and thus all the flagd documentation is also relevant. One specific callout is to point of the fact that `ResolveString` isn't the only option, but [other options are available](https://github.com/open-feature/flagd/blob/main/docs/usage/evaluation_examples.md).

Kavindu-Dodan commented:
> Thanks for these changes. They largely look great! A few comments though:
> 
> 1. `docs/getting_started.md` and `docs/quick_start.md` should be combined. They serve largely the same purpose and as such, having both is confusing and redundant.
> 2. At the end of `docs/installation.md` we should link to `docs/getting_started.md` otherwise, after installation, I don't know what else to do.
> 3. At the end of `docs/quick_start.md` (or wherever that content ends up, we should mention that the OFO uses flagd as a flag provider and thus all the flagd documentation is also relevant. One specific callout is to point of the fact that `ResolveString` isn't the only option, but [other options are available](https://github.com/open-feature/flagd/blob/main/docs/usage/evaluation_examples.md).

thank you @agardnerIT for the feedback. I addressed all your points 

1. Kept one quick start guide : this is now comprehensive and includes minimal installation instructions
2. Added the link to quick start guide : operator installation can be skipped but users can follow other steps to deploy a workload
3. Added the `what's next` section [1] linking to other resources 

[1] - https://github.com/open-feature/open-feature-operator/pull/496/files#diff-61f4196322cfa9f52c5f548f13b7f781243b2cae898862a8688462d914ba7173R178-R182

toddbaert commented on docs/README.md:
```diff
@@ -1,24 +1,41 @@
 
-Configuration of the deployed sidecars is handled through the `FeatureFlagConfiguration` CRs defined in the `openfeature.dev/featureflagconfiguration` annotation of a deployed `PodSpec`. 
-> Further configuration of the operator will be possible in the future, to help contribute [click here](https://github.com/open-feature/open-feature-operator/issues)
+Configuration of the deployed sidecars is handled through the `FeatureFlagConfiguration` custom resources defined in the `openfeature.dev/featureflagconfiguration` annotation of a deployed `PodSpec`.
```
I think this isn't really the best explanation.

`FeatureFlagConfiguration` actually is inteded most to define flags, while `FlagSourceConfiguration` controls the sidecars.

Kavindu-Dodan replied:
Thank you 🙌 

A good catch and I think naming is a little misleading and confusing sometimes. 

I updated the documentation to correct this

toddbaert commented on docs/quick_start.md:
```diff
@@ -0,0 +1,182 @@
+## Quick Start for OpenFeature Operator
```
I think this is a really nice addition. Using busy-box, having no ingress, etc, is nice b/c it's very "barebones".

Kavindu-Dodan replied:
Agree, we should keep it minimal and let users get things up and running fast. Later, we can refer them to detailed guides

toddbaert reviewed (changes requested):
This is an improvement, but I think there's a mistake [here](https://github.com/open-feature/open-feature-operator/pull/496/files#r1255857908).

Kavindu-Dodan commented:
> This is an improvement, but I think there's a mistake [here](https://github.com/open-feature/open-feature-operator/pull/496/files#r1255857908).

Thank you. I corrected them and also expanded `Flag Source configuration` section [1] with other flag sources and configuration options. Strangely, we never documented all the options we supported  

[1] - https://github.com/open-feature/open-feature-operator/blob/b7a6b531ed9ab6edfe5ecf6038bfdc2ae570a811/docs/flag_source_configuration.md

thisthat commented on docs/README.md:
```diff
@@ -1,24 +1,41 @@
+# Documentation
 
-This directory contains all OpenFeature Operator documentation, see table of contents below:
+This directory contain OpenFeature Operator documentation. 
```
```suggestion
This directory contains OpenFeature Operator documentation. 
```

toddbaert commented on docs/flag_source_configuration.md:
```diff
@@ -1,7 +1,96 @@
+
+`FlagSourceConfiguration` support multiple flag sources. Sources are configured as a list and given below are supported sources and their configurations,
+
+### kubernetes aka `FeatureFlagConfiguration`
```
both this, and `flagd-proxy` make use of `FeatureFlagConfiguration`. I think you might want to put these two closer together, and conceptually link and contrast them (probably just highlight that they both use `FeatureFlagConfiguration`, but one uses a proxy and doesn't require direct access of the workload to k8s).

Kavindu-Dodan replied:
Good point. Added a description and reordered the source list to keep k8s & proxy approach close to each other

toddbaert reviewed (approved):
LGTM, but I'd like to see this last small change: https://github.com/open-feature/open-feature-operator/pull/496/commits/1dbf3d0745bf0e33e6090297c4a46beb34b84389#r1258239251

toddbaert commented:
@Kavindu-Dodan the last set of changes looks great! I think after it's merged we could put @thisthat 's [changes](https://github.com/open-feature/open-feature-operator/pull/497) in/around these changes as a sub-point.

toddbaert reviewed (approved):
@Kavindu-Dodan the last set of changes looks great! I think after it's merged we could put @thisthat 's [changes](https://github.com/open-feature/open-feature-operator/pull/497) in/around these changes as a sub-point.

Kavindu-Dodan commented on docs/annotations.md:
```diff
@@ -25,6 +31,10 @@ Example:
+
+Given below are references to **deprecated** annotations used by previous versions of the operator.
+
 ### `openfeature.dev/allowkubernetessync`
```
Good catch, this is internal to the operator and worth documenting.

beeme1mr commented on docs/annotations.md:
```diff
@@ -26,9 +32,14 @@ Example:
 ### `openfeature.dev/allowkubernetessync`
-*This annotation is used internally by the operator.*  
-This annotation is used to mark pods which should have their permissions backfilled in the event of an upgrade. When the OFO manager pod is started, all `Service Accounts` of any `Pods` with this annotation set to `"true"` will be added to the `flagd-kubernetes-sync` `Cluster Role Binding`.
+*This annotation is used INTERNALLY by the operator.*  
```
```suggestion
*This annotation is used INTERNALLY by the operator.*
```
